packages/fxa-settings/src/pages/InlineTotpSetup/container.tsx

```diff
@@ -102,51 +157,17 @@ export const InlineTotpSetupContainer = ({
     [navTo, totp, signinState]
   );
 
-  useEffect(() => {
-    (async () => {
-      try {
-        // The user is navigated to this page by the web application in response to
-        // a sign-in attempt.  But let's do some sanity checks.
-        const sessionVerified = await session.isSessionVerified();
-        if (!sessionVerified) {
-          navTo('signin_token_code', signinState ? signinState : undefined);
-        }
-
-        if (totpStatus?.account.totp.verified) {
-          navTo('signin_totp_code', signinState ? signinState : undefined);
-        }
-
-        if (!totp) {
-          const totpResp = await createTotp({
-            variables: {
-              input: {
-                metricsContext,
-              },
-            },
-          });
-          setTotp(totpResp.data?.createTotp);
-        }
-      } catch (_) {
-        navTo('signup');
-      }
-    })();
-  }, [
-    isSignedIn,
-    signinState,
-    session,
-    navTo,
-    totp,
-    createTotp,
-    totpStatus?.account.totp.verified,
-    metricsContext,
-  ]);
-
-  if (totpStatusLoading || !totp) {
+  if (!isSignedIn || !signinState) {
     return <LoadingSpinner fullScreen />;
   }
 
-  if (!isSignedIn || !signinState) {
-    navTo('signup');
+  if (
```

Comment:
Could this be combined with L160-162?

Reply:
Ah sorry, I missed this ^. I'll file a small polish PR.